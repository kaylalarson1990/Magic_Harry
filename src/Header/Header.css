body {
    margin: 0;
    padding: 0;
    font-size: 15px;
  }
  nav {
    margin: 0;
    overflow: hidden;
  }
  nav ul {
    margin: 0;
    padding: 0;
    display: flex;
  }
  nav ul li {
    display: inline-block;
    list-style-type: none;
  }
  nav ul li a {
    color: black;
    display: block;
    line-height: 3em;
    padding: 1em 1em;
    text-decoration: none;
  }
  nav ul li:first-child {
    margin-right: auto;
  }
