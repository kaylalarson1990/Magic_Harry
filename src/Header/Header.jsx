import React, { Component } from "react";
import "./Header.css";
import { connect } from "react-redux";
import glasses from "../images/glasses.png";
import { NavLink } from "react-router-dom";

export class Header extends Component {
  render() {
    return (
      <header>
        <nav>
          <ul className="nav">
            <li>
              <a href="#" className="glasses">
                <NavLink to="/" exact activeClassName="active">
                  <img
                    src={glasses}
                    alt="harry potter glasses"
                    className="glasses"
                  />
                </NavLink>
              </a>
            </li>
            <li>
              <a href="#">
                <NavLink to="/characters" exact activeClassName="active">
                  CHARACTERS
                </NavLink>
              </a>
            </li>
            <li>
              <a href="#">
                <NavLink to="/spells" exact activeClassName="active">
                  SPELLS
                </NavLink>
              </a>
            </li>
            <li>
              <a href="#">
                <NavLink to="/houses" exact activeClassName="active">
                  HOUSES
                </NavLink>
              </a>
            </li>
            <li>
              <a href="#">
                <NavLink to="/favorites" exact activeClassName="active">
                  FAVORITES
                </NavLink>
              </a>
            </li>
          </ul>
        </nav>
      </header>
    );
  }
}

export const mapStateToProps = state => ({
  spells: state.spells,
  houses: state.houses,
  characters: state.characters
});

export default connect(mapStateToProps)(Header);
