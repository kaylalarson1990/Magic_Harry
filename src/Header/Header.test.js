import React from "react";
import { shallow } from "enzyme";
import {Header, mapStateToProps} from "./Header";

describe('Header', () => {
    let wrapper, props;
    beforeEach(() => {
        wrapper = shallow(<Header />)
    })

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})

describe("mapStateToProps", () => {
    it("should return an object with a house array", () => {
      const mockState = {
        houses: "Apartment complex",
        type: "SET_HOUSES"
      };
      const expected = {
        houses: "Apartment complex"
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });

    it("should return an object with a spell array", () => {
        const mockState = {
          spells: "Accio",
          type: "SET_SPELLS"
        };
        const expected = {
          spells: "Accio"
        };
        const mappedProps = mapStateToProps(mockState);
        expect(mappedProps).toEqual(expected);
      });

      it("should return an object with a characters array", () => {
        const mockState = {
          characters: "Kayla",
          type: "SET_CHARACTERS"
        };
        const expected = {
          characters: "Kayla"
        };
        const mappedProps = mapStateToProps(mockState);
        expect(mappedProps).toEqual(expected);
      });
  });