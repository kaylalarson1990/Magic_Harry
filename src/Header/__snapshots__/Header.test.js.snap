// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header>
  <nav>
    <ul
      className="nav"
    >
      <li>
        <a
          className="glasses"
          href="#"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/"
          >
            <img
              alt="harry potter glasses"
              className="glasses"
              src="glasses.jpg"
            />
          </NavLink>
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/characters"
          >
            CHARACTERS
          </NavLink>
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/spells"
          >
            SPELLS
          </NavLink>
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/houses"
          >
            HOUSES
          </NavLink>
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/favorites"
          >
            FAVORITES
          </NavLink>
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
